src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -293,10 +293,12 @@ def _parse_file_name(self, path):
         stream = get_entry(matched, "scan_id")
         file_idx = int(get_entry(matched, "file_idx_in_scan"))
         try:
-            idx_offset = get_entry(matched, "file_idx_offset")
+            idx_offset = int(get_entry(matched, "file_idx_offset"))
             file_idx += idx_offset
         except utils.UsageError:
             pass
```

Comment:
I find the meaning of  `UsageError` difficult to understand (why is this not `KeyError`?).
```suggestion
            # The file_regex doesn't contain a file_idx_offset group and
            # therefore no offset should be added
            pass
```